packages/@aws-cdk/aws-route53/lib/records/alias.ts

```diff
@@ -0,0 +1,53 @@
+import { CloudFrontWebDistribution } from '@aws-cdk/aws-cloudfront';
+import { BaseLoadBalancer } from '@aws-cdk/aws-elasticloadbalancingv2';
+import { Construct } from '@aws-cdk/cdk';
+import { HostedZoneRef } from '../hosted-zone-ref';
+import { cloudformation } from '../route53.generated';
+import { determineFullyQualifiedDomainName } from './_util';
+
+export interface AliasRecordProps {
+  hostedZone: HostedZoneRef | string;
+  recordName: string;
+  target: CloudFrontWebDistribution | BaseLoadBalancer;
```

Comment:
If both need to be a potential target, the only way to do that in the absence of type unions is an interface:

```ts
export class IAliasRecordTarget {
   asAliasRecordTarget(): AliasRecordTargetProps;
}

export interface AliasRecordTargetProps {
  hostedZoneId: string;
  dnsName: string;
}
```

And make `CloudFrontWebDistribution` and `BaseLoadBalancer` both implement that interface.

(And you would invert the dependencies in `package.json`. If you do that and `lerna bootstrap/lerna link`, you will be hit with an infinite recursion error on the next build, and that'll be because the symlinks are still there from your previous dependency direction. Remove the stale but unremoved dependency links from `aws-route53/node_modules/@aws-cdk` to continue)

Reply:
Makes sense